### PR TITLE
AAP-5182 - add content for RHSSO on controller for RH OCP

### PR DESCRIPTION
**Pull request by dcdacosta**
This PR adds content requested in https://issues.redhat.com/browse/AAP-5182 and includes the following:

- New Setting up authentication chapter with instructions for setting RH-SSO on controller

Preview link (VPN required): http://file.rdu.redhat.com/~ddacosta/AAP5182/master.html

### CODE REVIEW

File: downstream/assemblies/platform/assembly-aap-os-rhsso.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+ifdef::context[:parent-context: {context}]
+
+[id="aap-os-rhsso"]
+
+:context: aap-authentication
+
+= Setting up authentication
```
[the remark this answers is not in the document]
Comment (dcdacosta) [reply downstream/assemblies/platform/assembly-aap-os-rhsso.adoc]: @rooftopcellist , I worry that changing it simply to "Setting up RHSSO" might be confusing with the other chapter about RHSSO on Hub; so what do you think of "Red Hat Single Sign-on integration with automation controller" which is in line with the blog title we used to develop the bulk of the content?

File: downstream/modules/platform/proc-configure-saml-controller.adoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+-----
++
+. Use the *SAML Team Map* field to map team members (users) from social authorization accounts.
+. Use the *SAML Team Attribute Mapping* field to translate user team membership.
+. Use the *SAML Security Config* field to define key value pairs that are passed to the underlying python-saml security setting.
+. Use the *SAML Service Provider Extra Configuration Data* field to define key value pairs to be passed to the underlying python-saml Service Provider configuration setting.
+. Use the *SAML IDP to EXTRA_DATA Attribute Mapping* to define a list of tuples that maps IDP attributes to extra_attributes. Each attribute will be a list of values, even if only 1 value.
+. Use the *SAML User Flags Attribute Mapping* field to map super users and system auditors from SAML.
+. Click btn:[Save].
```
Comment (john-westcott-iv) [inline downstream/modules/platform/proc-configure-saml-controller.adoc:96]: Do we want to add any details around these maps?

File: downstream/modules/platform/proc-create-client.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+[role=_abstract]
+Use the following procedure to create a SAML client.
+.Procedure
+. Log into {RHSSO} as the admin user.
+. Go to your {PlatformName} realm.
+. Navigate to menu:Clients[Create].
+. Click *select file* to import the data that exists on {PlatformName}. To get the configuration, run the following command:
+-----
+curl -L -k https://tower.usersys.redhat.com/sso/metadata/saml/ > rhsso-metadata.xml
```
Comment (john-westcott-iv) [inline downstream/modules/platform/proc-create-client.adoc:13]: This command might change if you don't name your SAML provider rhsso.

File: downstream/modules/platform/proc-create-mappers.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
+[id="configure-mappers"]
+
+= Create mappers on the {PlatformNameShort} client
+
+[role=_abstract]
+The final step to take is to create the mappers on the {PlatformNameShort} {RHSSO} client. This defines the information coming from {RHSSO} and maps it to your {PlatformNameShort} users.
```
Comment (john-westcott-iv) [inline downstream/modules/platform/proc-create-mappers.adoc:6]: The mapping is just in RHSSO, so this should read:
The final step to take is to create the mappers on the {RHSSO} client.

**Comment by john-westcott-iv** [comment]
Also, I would like to again throw out that I feel like if we put this in official docs we should have someone from the RHSSO team review our "suggestions" to customers and pick it apart. There are a lot of settings in RHSSO and I want to make sure we are not telling customers to do something that goes against security standards.

Or we could put in the document somewhere that this is a sample configuration for testing only.

**Comment by dcdacosta** [comment]
This document has been converted to a KCS article; therefore, I'm closing this PR. 

KCS article draft is available here: https://access.redhat.com/articles/7028643

This article will be published once technical reviews are complete.

**Pull request closed without merging**